tictactoe/src/Tictactoe.java

```diff
@@ -0,0 +1,238 @@
+import java.util.Scanner;
+
+public class Tictactoe {
+    public static void main(String[] args) {
+
+        int[][] array = {{2, 2, 2}, {2, 2, 2}, {2, 2, 2},}; //2 клетка свободна, 1 - крестик Х, 0 - нолик 0
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            out(array); //Вывод игрового поля
+            userInput(sc, array); //Ход игрока
+            if (!stepAI(array, 0, 2)) {     //Сначала ищем 0 в количестве 2 шт.
+                if (!stepAI(array, 1, 2)) { //Если не находим, то ищем 1 в количестве 2 шт
+                    stepAI(array, 1, 1);    //Если не находим, то ищем 1 в количестве 1 шт
+                }
+            }
+            if (check(array)) { //Проверка поля на выйгрыш либо наличие своьодных клеток
+                out(array);
+                System.out.println("Игра окончена");
+                break;
+            }
+        }
+    }
+
+    //Вывод игрового поля
+    static void out(int[][] array) {
+        System.out.println("|---|---|---|");
+        for (int[] ints : array) {
+            for (int j = 0; j < array.length; j++) {
+                if (ints[j] == 0) {
+                    System.out.print("| 0 ");
+                }
+                if (ints[j] == 1) {
+                    System.out.print("| X ");
+                }
+                if (ints[j] == 2) {
+                    System.out.print("|   ");
+                }
+            }
+            System.out.println("|");
+            System.out.println("|---|---|---|");
+
+        }
+    }
+
+    //Ход игрока
+    static void userInput(Scanner sc, int[][] array) {
+
+        if (check(array)) { //Проверка количества оставшихся незаполненных клеток
+            return;
+        }
+
+        int i, j; //Строка и столбец для хода игрока
```

Comment:
if you know that it is row and column you can give this variables correct name and no comment needed.

Reply:
> if you know that it is row and column you can give this variables correct name and no comment needed.

Corrected according to remark